smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/ProtocolGenerator.java

```diff
@@ -119,6 +121,32 @@ default ApplicationProtocol resolveApplicationProtocol(
      */
     void generateResponseDeserializers(GenerationContext context);
 
+    /**
+     * Generates the name of a serde function for shapes of a service.
+     *
+     * @param symbol The symbol the serde function is being generated for.
+     * @param protocol Name of the protocol being generated.
+     * @param isInput Whether the function is for input or output.
+     * @return Returns the generated function name.
+     */
+    static String getSerdeFunctionName(Symbol symbol, String protocol, boolean isInput) {
```

Comment:
Consider just having two method: getSerFunctionName and getDeserFunctionName. Also, should this actually be a public API, or is this implementation specific to HTTP bindings?